File: Exemplos/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/SAT/CancelarUltimaVenda.ts
import {SatCommand} from './SatCommand';

export class CancelarUltimaVenda extends SatCommand {
  readonly numSessao: number;
  readonly codAtivacao: string;
  readonly numeroCFe: string;
  readonly dadosCancelamento: string;

  constructor(
    numSessao: number,
    codAtivacao: string,
    numeroCFe: string,
    dadosCancelamento: string,
  ) {
    super('CancelarUltimaVenda');
    this.numSessao = numSessao;
    this.codAtivacao = codAtivacao;
    this.numeroCFe = numeroCFe;
    this.dadosCancelamento = dadosCancelamento;
  }

  functionParameters(): object {
    return {
      numSessao: this.numSessao,
      codAtivacao: this.codAtivacao,
      numeroCFe: this.numeroCFe,
      dadosCancelamento: this.dadosCancelamento,
    };
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/IntentServices/Sat/CancelarUltimaVenda.cs
namespace Xamarin_Forms_Intent_Digital_Hub.IntentServices.Sat
{
    class CancelarUltimaVenda : SatCommand
    {
        readonly int numSessao;
        readonly string codAtivacao;
        readonly string numeroCFe;
        readonly string dadosCancelamento;

        public CancelarUltimaVenda(int numSessao, string codAtivacao, string numeroCFe, string dadosCancelamento) : base("CancelarUltimaVenda")
        {
            this.numSessao = numSessao;
            this.codAtivacao = codAtivacao;
            this.numeroCFe = numeroCFe;
            this.dadosCancelamento = dadosCancelamento;
        }

        protected override string FunctionParameters()
        {
            return "\"numSessao\"" + ":" + numSessao + "," +
                    "\"codAtivacao\"" + ":" + "\"" + codAtivacao + "\"" + "," +
                    "\"numeroCFe\"" + ":" + "\"" + numeroCFe + "\"" + "," +
                    "\"dadosCancelamento\"" + ":" + "\"" + dadosCancelamento + "\"";
        }
    }
}